### PR TITLE
Feature/combine eth erc20

### PR DESCRIPTION
**Pull request by cwhinfrey**
- Combine ETH and ERC20 L1 bridges
- Forward value to Arbitrum through MessengerWrapper

### CODE REVIEW

**Comment by cwhinfrey** [comment]
Builds on top of #79

File: contracts/bridges/L1_Bridge.sol [hunk 1]
```diff
@@ -357,6 +381,25 @@ abstract contract L1_Bridge is Bridge {
...
+            (bool success, ) = recipient.call{value: amount}(new bytes(0));
+            require(success, 'L1_ETH_BRG: ETH transfer failed');
+        } else {
+            IERC20(token).safeTransfer(recipient, amount);
+        }
+    }
+
+    function _transferToBridge(address from, uint256 amount) internal override {
+        if (token == ETH_ADDRESS) {
+            require(msg.value >= amount, "L1_ETH_BRG: Value is less than amount");
```
Comment (shanefontaine) [inline contracts/bridges/L1_Bridge.sol:397]: I believe this should be kept as `msg.value == amount` -- I don't see where having it `>=` would be beneficial.

 In `sendToL2`, for example, the `msg.value` should already have been transferred to the contract before the execution of `_transferToBridge`.

**Review by shanefontaine — CHANGES_REQUESTED** [review]

File: contracts/bridges/L1_Bridge.sol (hunk 1 above)
Comment (cwhinfrey) [reply contracts/bridges/L1_Bridge.sol:397]: This is needed if we want to forward additional ETH to Arbitrum through the message wrapper

Comment (shanefontaine) [reply contracts/bridges/L1_Bridge.sol:397]: You are correct. I was thinking this was sending ETH but instead this is just checking the values. I think this is valid 👍

**Pull request merged**